### ISSUE TITLE
Error: The terraform-provider-google_v4.43.0_x5 plugin crashed!

### ISSUE THREAD

**Issue by pusnaik2016** [terraform-google-modules/terraform-google-network#403]
### TL;DR

Error: The terraform-provider-google_v4.43.0_x5 plugin crashed!

### Expected behavior

terraform plan

The output should look like this:

...
Plan: 4 to add, 0 to change, 0 to destroy.

### Observed behavior

student_03_c510f1c87e7d@cloudshell:~/tfinfra (qwiklabs-gcp-03-f90c018e8b30)$ terraform plan
╷
│ Error: Plugin did not respond
│
│   with provider["registry.terraform.io/hashicorp/google"],
│   on provider.tf line 1, in provider "google":
│    1: provider "google"  {}
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-google_v4.43.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 6 [running]:
github.com/hashicorp/terraform-provider-google/google.GetCurrentUserEmail(0x32373f8?, {0xc0001d61c0?, 0x322e820?})
        github.com/hashicorp/terraform-provider-google/google/utils.go:511 +0x125
github.com/hashicorp/terraform-provider-google/google.(*Config).logGoogleIdentities(0xc0008bf000)
        github.com/hashicorp/terraform-provider-google/google/config.go:589 +0xd4
github.com/hashicorp/terraform-provider-google/google.(*Config).LoadAndValidate(0xc0008bf000, {0x32373f8?, 0xc000cc6040})
        github.com/hashicorp/terraform-provider-google/google/config.go:480 +0x265
github.com/hashicorp/terraform-provider-google/google.providerConfigure({0x32374a0, 0xc0009fe0c0}, 0x2c9e360?, 0x1?)
        github.com/hashicorp/terraform-provider-google/google/provider.go:1627 +0x2a05
github.com/hashicorp/terraform-provider-google/google.Provider.func1({0x32374a0?, 0xc0009fe0c0?}, 0xc0009fe0c0?)
        github.com/hashicorp/terraform-provider-google/google/provider.go:905 +0x27
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc000089b60, {0x32374a0, 0xc0009fe0c0}, 0xd?)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/provider.go:297 +0x1e6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc0000ad3e0, {0x32374a0?, 0xc0009d8720?}, 0xc000d90df8)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:557 +0x345
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).Configure(0xc000127360, {0x32374a0?, 0xc0009d8120?}, 0xc000c57cc0)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:556 +0x2ce
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_Configure_Handler({0x2c63920?, 0xc000127360}, {0x32374a0, 0xc0009d8120}, 0xc0009cc150, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:331 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000001e0, {0x323c888, 0xc0008b4000}, 0xc00091aea0, 0xc000846db0, 0x438de18, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0000001e0, {0x323c888, 0xc0008b4000}, 0xc00091aea0, 0x0)
        google.golang.org/grpc@v1.50.1/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.50.1/server.go:963 +0x28a

Error: The terraform-provider-google_v4.43.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Terraform Configuration

```hcl
Didn't get this details
```

### Terraform Version

```sh
terraform version
Terraform v1.3.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.43.0
```

### Additional information

NA

**Comment by bharathkkb** [comment]
Thanks for the report! This is an upstream provider issues https://github.com/hashicorp/terraform-provider-google/issues/13051
